parquet/src/file/serialized_reader.rs

```diff
@@ -452,21 +452,20 @@ pub(crate) fn decode_page(
     // maximum page header size and abort if that is exceeded.
     let buffer = match decompressor {
         Some(decompressor) if can_decompress => {
-            let uncompressed_size = page_header.uncompressed_page_size as usize;
-            let mut decompressed = Vec::with_capacity(uncompressed_size);
-            let compressed = &buffer.as_ref()[offset..];
+            let uncompressed_page_size = page_header.uncompressed_page_size as usize;
```

Comment:
If we are going to be messing with this code anyways, maybe we can do a checked conversion from signed to unsigned as well:

```suggestion
            let uncompressed_page_size = usize::try_from(page_header.uncompressed_page_size)?;
```